systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaNodePoolResource.java

```diff
@@ -111,4 +115,19 @@ public static KafkaNodePool convertKafkaResourceToKafkaNodePool(Kafka resource)
 
         return builder.build();
     }
+
+    public static LabelSelector getKafkaNodePoolLabelSelector(String kafkaClusterName, String kafkaNodePoolName) {
+        Map<String, String> matchLabels = new HashMap<>();
+        matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, kafkaClusterName);
+        matchLabels.put(Labels.STRIMZI_KIND_LABEL, Kafka.RESOURCE_KIND);
+        matchLabels.put(Labels.STRIMZI_CONTROLLER_NAME_LABEL, KafkaNodePoolResource.getStrimziPodSetName(kafkaClusterName, kafkaNodePoolName));
+        return new LabelSelectorBuilder()
+            .withMatchLabels(matchLabels)
+            .build();
+    }
```

Comment:
I'm not against this method, for mine upcoming PR I have following:

```
    public static LabelSelector getLabelSelector(String clusterName, String poolName, ProcessRoles processRole) {
        Map<String, String> matchLabels = new HashMap<>();
        matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, clusterName);
        matchLabels.put(Labels.STRIMZI_KIND_LABEL, Kafka.RESOURCE_KIND);
        matchLabels.put(Labels.STRIMZI_POOL_NAME_LABEL, poolName);

        switch (processRole) {
            case BROKER -> matchLabels.put(Labels.STRIMZI_BROKER_ROLE_LABEL, "true");
            case CONTROLLER -> matchLabels.put(Labels.STRIMZI_CONTROLLER_LABEL, "true");
            default -> {}
        }

        return new LabelSelectorBuilder()
            .withMatchLabels(matchLabels)
            .build();
    }
```

so maybe instead of the `STRIMZI_CONTROLLER_NAME_LABEL` I would use the `STRIMZI_POOL_NAME_LABEL`, so you will not need the helpers method for getting the SPS name